pyproject.toml

```diff
@@ -147,7 +147,7 @@ synapse_review_recent_signups = "synapse._scripts.review_recent_signups:main"
 update_synapse_database = "synapse._scripts.update_synapse_database:main"
 
 [tool.poetry.dependencies]
-python = "^3.7.1"
+python = "^3.8.0"
```

Comment:
https://github.com/matrix-org/synapse/actions/runs/5416421916/jobs/9846127375?pr=15851
```
Dependency walk failed at importlib_metadata (>=1.4)
pyproject.toml was updated without running `poetry lock --no-update`.
```

Not sure what to do since I did run `poetry lock --no-update` and running it now shows no changes :face_exhaling:

Reply:
It looks like `importlib_metadata` is a Python <3.8 dependency. I've removed it and now everything is happy :tada: 

https://github.com/matrix-org/synapse/blob/13fc89148c37f6c653dd237377ff165249fa73cc/pyproject.toml#L206-L208